check_os.py

```diff
@@ -17,53 +17,30 @@
                         "Do you want to stop the process and do a clean restart instead?"
 
 
+def _do_popup(title, text):
+    """
+    Create a native pop-up without any third party dependency.
+
+    :param title: the pop-up title
+    :param text: the pop-up body
+    """
+    try:
+        import win32api
+        win32api.MessageBox(0, text, title)
+    except ImportError:
+        os.system('xmessage -center "$(printf "%s")"' % (text))
```

Comment:
I would like to point out that this appears to be a security risk. At least a ticking time-bomb. If an attacker can control the content of `text`, any arbitrary shell command can be executed.

Reply:
using `os.subprocess` with `shell=False` is probably sufficient to fix it btw

Reply:
I confirm that using `os.system` is a security risk and is considered to be a bad practice these days. `os.subprocess` with `shell=False` is the way to go safely.

Reply:
Thanks for your feedback! I will change this as part of my PR (#4815)